### DISCUSSION TITLE
strange behavior with "ctrl+alt" combinations on my laptop

### DISCUSSION THREAD

**Discussion by Monkjj** [autokey/autokey#690] [Q&A]
First of all, a big thanks to anybody who contributed to this amazing project! Still, I think autokey might cause problems on my computer. They might come from the way I've configured some shortcuts but I'm not sure about that.

A while ago, I noticed a strange behavior when trying to open the terminal via "ctrl+alt+t" on my Linux Mint 20.3 laptop. Actually, when I press any key-combination that involves "ctrl" and "alt", my mouse jumps several times within a few seconds and the action that the shortcut should actually trigger doesn't happen. This happens when autokey is activated and when it's not activated (i.e. when I "quit" via the panel). But when autokey is activated, something with my open windows changes. Sometimes they lose the "minimize", "maximize" and "close" buttons or they just change their position or both. Also, when they lose their buttons, moving a window (i.e. Terminal) over another open window, doesn't change the focus. The moved window doesn't show before the other. Usually, I am able to restore the buttons and everything using the "ctrl+alt" combination again. However, the "ctrl+alt+backspace" combination doesn't show this behavior and terminates the server properly.

I do have hotkeys configured that involve "ctrl+alt" and that include mouse actions. But the behavior also shows when I press a combination that I did not specify in my custom scripts (i.e. "ctrl+alt+t")

I thought maybe it's something about linux and tried the suggestions from [this question](https://superuser.com/questions/1348136/ctrlaltshift-shortcuts-not-working-on-linux-mint-19), e.g. 
`sudo add-apt-repository ppa:nrbrtx/xorg-hotkeys`
`sudo apt update`
`sudo apt dist-upgrade`

but it didn't change the weird behavior. My output of `grep "_alt_" /usr/share/X11/xkb/rules/xorg` is:

`  altwin:ctrl_alt_win	=	+altwin(ctrl_alt_win)`
`  altwin:swap_alt_win	=	+altwin(swap_alt_win)`
`  grp:ctrl_alt_toggle	=	+group(ctrl_alt_toggle)`
`  grp:ctrl_alt_toggle_bidir	=	+group(ctrl_alt_toggle_bidir)`
`  terminate:ctrl_alt_bksp =	+terminate(ctrl_alt_bksp)`

I'm posting this question here because I assume autokey might be involved in this. I am aware, that something unusual also happens when autokey is closed but it gets way worse when autokey is on. So maybe you have some suggestions that I could try to fix this problem?

**Comment by josephj11** [comment]
Comments:
If you boot your system without AutoKey running, do you get any of this strange behavior or does it only start after AutoKey is running or was running and was terminated?

If at least some of this problem is independent of AutoKey, it would make sense to address that first in the most minimal configuration that produces the problems.

Is this a recent development? Was everything fine before some upgrade, etc.?

If you have the time, you can fire up old versions of things in a VM and possibly narrow down the time frame or source of the problem. Same goes if you want to explore other distros, although with them, you can dual boot or run from a flash drive installation.

I have no idea what any of the Xorg stuff is about, but on kubuntu 18.04, the output of that command appears to be identical to what you see, so that's probably unrelated.

I looked through my scripts and I have one triggered by Ctrl+Alt+t and another triggered by Ctrl+Alt+p and I haven't seen any problems. I use the first one occasionally and the other one was for testing something and doesn't get used.

What does your desktop use to open a terminal window? I seem to remember Ctrl+Alt+t being assigned to that and the desktop almost always sees and consumes hotkeys before AutoKey gets to see them.

We have had a number of reports related to Linux Mint lately, but have not analysed them.

I'm not sure it will help in this case, but when AutoKey appears to be misbehaving, it's often useful to run a [trace](https://github.com/autokey/autokey-python2/wiki/Problem-Reporting-Guide) to see things from its point of view.

It might be useful to continue this discussion on Gitter where the devs and advanced users hang out. It was fine to post it here, but there are more people there. See boiler plate below.

----

BTW: Whenever you have a question (as opposed to an issue) it's better to post it on our [list](https://groups.google.com/forum/#!forum/autokey-users) or on [Gitter](https://gitter.im/autokey/autokey). The list is more user oriented whereas more technical questions will get a better answer on Gitter.

Both of these venues have a number of active users, so you are likely to get more, better, and faster answers on them than here on GitHub where only our very busy developers and a few users see the posts. We also prefer to have the developers developing rather than taking time to answer questions and manage issues that the user community could handle for them.
